portal/i18next_gulpfile.js

```diff
@@ -0,0 +1,186 @@
+ */
+var gulp = require('gulp');
+var source = require('vinyl-source-stream');
+var request = require('request');
+var merge = require('merge2');
+var concat = require('gulp-concat');
+var buffer = require('gulp-buffer');
+var del = require('del');
+var uglify = require('gulp-uglify');
+var rename = require('gulp-rename');
+var scanner = require('i18next-scanner');
+var concatPo = require('gulp-concat-po');
+const path = require('path');
+const fs = require('fs');
+const i18nextConv = require('i18next-conv');
+/*
+ * where the generated json/pot files from extraction of js files will reside
+ */
+const translationSourceDir = path.join(__dirname, './translations/js/src/');
+/*
+ * the path to the converted json file from po file of corresponding locale
+ * JS files will consume the translated text from here
+ * note json files are saved for each specific locale
+ */
+const translationDestinationDir = path.join(__dirname,'./static/files/locales/');
+/*
+ * supported languages
+ */
+const languagesArray = ['en-US', 'en-AU'];
+/*
+ * namespace
+ */
+const nameSpace = "translation";
+
+/*
+ * helper function for writing file
+ */
+function save(target) {
+  return result => {
+    fs.writeFileSync(target, result);
+  };
+};
+
+/*
+ * extracting text from js/html files into json file
+ */
+gulp.task('i18next-extraction', function() {
+    console.log("extracting text and generate json file ...")
+    del(['translations/js/src/translation.json']);
+    return gulp.src(['static/**/*.{js,html}'])
+               .pipe(scanner({
+                    lngs: languagesArray, // supported languages
+                    keySeparator: "|",
+                    nsSeparator: "|",
+                    attr: {
+                        list: ['data-i18n'],
+                        extensions: ['.js', '.html', '.htm']
+                    },
+                    func: {
+                        list: ['i18next.t', 'i18n.t'],
+                        extensions: ['.js', '.jsx']
+                    }
+                    ,
+                    resource: {
+                        //the source path is relative to current working directory as specified in the destination folder
+                        savePath: './src/' + nameSpace + '.json'
+                    }
+                }))
+              .pipe(gulp.dest('translations/js'));
+});
+
+/*
+ * convert json to pot (the definition file) for translator's consumption
+ */
+gulp.task('i18nextConvertJSONToPOT', ['i18next-extraction'], function() {
+
+    const options = {/* you options here */}
+   /*
+    * converting json to pot
+    */
+    console.log("converting JSON to POT...");
+    del(['translations/js/src/translation.pot']);
+    i18nextConv.i18nextToPot('en', fs.readFileSync(translationSourceDir+nameSpace+".json"), options).then(save(translationSourceDir+nameSpace+".pot"));
+
+});
+
+/*
+ * combine newly created pot file to existing messages.pot file ???
+ * do we need this step??
+ */
```

Comment:
@ivan-c Do we want to merge the POT files here? It might be better to keep them separate, right?

I guess the main question is, how does this process fit in with our existing workflow? Should they be integrated (aka we'd want run this and upload the resulting POT file to smartling as part of our `manage.translation_upload()` process), or separate (in which case I'd imagine we'd want to do a separate round of uploading/downloading here)?

Reply:
I'm tempted to keep them separate, (maybe `portal/translations/frontend.pot`?) at least for now, if only because it would be more work to separate them later if we changed our minds.

We'll still want to merge all the PO files for the same language, but I think that makes sense to do after downloading them from Smartling.

Thoughts?

Reply:
separate sounds good.  How you are guys handling this in CPRO?  I am curious..is it only a separate pot file for front end as well?

Reply:
We've done it in a hacky way- [one JS file per language](https://gitlab.cirg.washington.edu/svn/dhair2/tree/master/app/webroot/js/localization), would be nice to switch to i18next there too.